src/sentry/runner/commands/configoptions.py

```diff
@@ -0,0 +1,150 @@
+    click.echo(get(key))
+
+
+@configoptions.command()
+@configuration
+def list():
+    "Fetches all options."
+    from sentry import options
+
+    for opt in options.all():
+        # click.echo(f"{opt.name} ({opt.type}) = {options.get(opt.name)}")
+        click.echo(f"{opt.name}: {options.get(opt.name)}")
+
+
+@configoptions.command()
+@click.argument("filename", required=True)
+@click.option(
+    "--dryrun",
+    is_flag=True,
+    default=False,
+    required=False,
+    help="Output exactly what changes would be made and in which order.",
+)
+@configuration
+def patch(filename, dryrun):
+    "Updates, gets, and deletes options that are each subsectioned in the given file."
+    import yaml
+
+    if dryrun:
+        click.echo("Dryrun flag on. ")
+    with open(filename) as stream:
+        configmap_data = yaml.safe_load(stream)
+        data = configmap_data.get("data", {}).get("options-patch.yaml", "")
+
+        keysToFetch = data.get("fetch", {})
+        keysToFetch = (line for line in keysToFetch.split("\n") if line)
+
+        keysToUpdate = data.get("update", {})
+        keysToUpdate = (line.split(": ") for line in keysToUpdate.split("\n") if line)
+
+        keysToDelete = data.get("delete", {})
+        keysToDelete = keysToFetch = (line for line in keysToDelete.split("\n") if line)
+
+        for key in keysToFetch:
+            click.echo(get(key, dryrun))
+
+        for key, val in keysToUpdate.items():
+            click.echo(set(key, dryrun))
+
+        for key in keysToDelete:
+            click.echo(delete(key, dryrun))
+
+
+@configoptions.command()
+@click.argument("filename", required=True)
+@click.option(
+    "--dryrun",
+    is_flag=True,
+    default=False,
+    required=False,
+    help="Output exactly what changes would be made and in which order.",
+)
+@configuration
+def strict(filename, dryrun):
+    "Deletes everything not in the uploaded file, and applies all of the changes in the file."
+    import yaml
+
+    from sentry import options
+
+    if dryrun:
+        click.echo("Dryrun flag on. ")
+
+    with open(filename) as stream:
+        configmap_data = yaml.safe_load(stream)
+
+        data = configmap_data.get("data", {}).get("options-strict.yaml", "")
+
+        kv_generator = (line.split(": ") for line in data.split("\n") if line)
+
+        db_keys = (opt.name for opt in options.all())
+
+        # update the database to remove all keys NOT in the given file.
+        for key in db_keys:
+            if key not in kv_generator:
+                click.echo(delete(key, dryrun))
+
+        for line in data.split("\n"):
+            if line:
+                line_tuple = line.split(": ")
+                key = line_tuple[0]
+                val = ""
+                if len(line_tuple) > 1:
+                    val = line_tuple[1]
+                click.echo(set(key, val, dryrun))
+
+
+def get(key, dryrun=False):
+    from sentry import options
+    from sentry.options.manager import UnknownOption
```

Comment:
If these are not command you expose you should not import what they need here. Ideally you would import everything at the top of the module. No need to go through the import step again every time you invoke the method. 
Now you cannot do that in the entry point of your application because you need settings to be imported as well. But that does not apply to any method which is not the entry point of the application.

Reply:
These methods can be called on their own as well from the command line with `sentry configoptions set [key] [value]` currently so they'll need these imports, I think. If that changes, I'll definitely remove them.

Reply:
If you want these to be called as CLI command, that's all good, though I think you have to add the `@command` annotation.

Reply:
👍